src/detectors/Python/PythonLanguageDetector.ts

```diff
@@ -0,0 +1,36 @@
+import { ILanguageDetector } from '../ILanguageDetector';
+import { injectable, inject } from 'inversify';
+import { LanguageAtPath, ProgrammingLanguage } from '../../model';
+import { IFileInspector } from '../../inspectors/IFileInspector';
+import { Types } from '../../types';
+import { fileNameRegExp, fileExtensionRegExp, sharedSubpath } from '../utils';
+import { uniq } from 'lodash';
+import * as nodePath from 'path';
+
+@injectable()
+export class PythonLanguageDetector implements ILanguageDetector {
+  private fileInspector: IFileInspector;
+  constructor(@inject(Types.IFileInspector) fileInspector: IFileInspector) {
+    this.fileInspector = fileInspector;
+  }
+
+  async detectLanguage(): Promise<LanguageAtPath[]> {
+    const result: LanguageAtPath[] = [];
+    const requirementsFiles = await this.fileInspector.scanFor(fileNameRegExp('requirements.txt'), '/');
+    const pyFiles = await this.fileInspector.scanFor(fileExtensionRegExp(['py', 'py3']), '/');
```

Comment:
`scanFor` is quite an expensive operation. I'd prefer to move it to the `else` block before the line 26. It's not necessary to scan for py files if some requirements files have been found.